app/jobs/split_pdf_job.rb

```diff
@@ -46,13 +46,18 @@ def perform(exam_template, path, original_filename=nil, current_user=nil)
         if m.nil?
           new_page.save File.join(error_dir, "#{filename}-#{i}.pdf")
           num_pages_qr_scan_error += 1
+          status = 'ERROR: QR code not found'
+          m_logger.log(status)
+          SplitPage.create(filename: filename, raw_page_number: i + 1, status: status)
```

Comment:
You're jumping through hoops here because `SplitPdfLog` is currently created after the `SplitPage`. You should do this the other way around - create the `SplitPdfLog` at the very beginning, and then create the `SplitPage` objects as an association.

Create the groups earlier too. This way you can populate all the `SplitPage` fields and save them immediately, with no subsequent update.